staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3860,21 +3860,33 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 		}
 	} else {
 		klog.V(4).Infof("service %v does not need custom health checks", apiService.Name)
-		// We only configure a TCP health-check on the first port
-		var tcpHealthCheckPort int32
-		for _, listener := range listeners {
-			if listener.InstancePort == nil {
-				continue
-			}
-			tcpHealthCheckPort = int32(*listener.InstancePort)
-			break
-		}
 		annotationProtocol := strings.ToLower(annotations[ServiceAnnotationLoadBalancerBEProtocol])
 		var hcProtocol string
+		var tcpHealthCheckPort int32
 		if annotationProtocol == "https" || annotationProtocol == "ssl" {
 			hcProtocol = "SSL"
+			for _, listener := range listeners {
+				if listener.InstancePort == nil {
+					continue
+				}
+				protocol := listener.Protocol
```

Comment:
Now that you point this out, it does kind of feel like `listener.InstanceProtocol` should be getting used here, but the docs on the type are a bit confusing so I'm not sure: https://docs.aws.amazon.com/sdk-for-go/api/service/elb/#Listener

Reply:
this indeed should be listeners.InstanceProtocol. 
Also, this will break when `ServiceAnnotationLoadBalancerBEProtocol` is set to HTTPS, but no certificated is used. (I'm not sure there there are ppl have such configuration conbination)